crates/bevy_ecs/src/system/query.rs

```diff
@@ -380,29 +382,37 @@ impl<'w, 's, D: QueryData, F: QueryFilter> Query<'w, 's, D, F> {
     #[inline]
     pub(crate) unsafe fn new(
         world: UnsafeWorldCell<'w>,
-        state: &'s QueryState<D, F>,
+        state: Entity,
         last_run: Tick,
         this_run: Tick,
     ) -> Self {
-        state.validate_world(world.id());
+        let query_state = world.world_mut().get::<QueryState<D, F>>(state).unwrap();
```

Comment:
It feels like transforming this to use `EntityRef` shouldn't be too difficult since the `Query` should own the `Entity`.

Reply:
`EntityRef` requires read access to all other components of that `Entity` as well, which will make it conflict with any other mutable query.

Reply:
I missed that this didn't namespace the entities it used at all...

I suspect this kind of work will require name-spacing of some kind to avoid the conflict.

After all if you have a Query that returns all Queries (assuming they are type erased) what sense does that make?

Reply:
It doesn't really make sense, but the way system params access currently works you can only request access for all components with a given id/type on entities that have or not some other components.

This will definitely need something like namespacing otherwise `Query<EntityMut>` would not work.

Reply:
I am just saying that caution should be taken to avoid expanding existing "everything" definitions ala `EntityRef`/`EntityMut` to suddenly block infrastructure bits from working. Those things didn't overlap before they would only overlap in the entity backed world if we didn't keep them distinct.

Solutions could be a separate world for an extreme example or segregating things so that the system knows that a query entity doesn't have a component on it that a user would add.

The main reason I brought up `EntityRef` is because it seems to more closely match what was written here. I guess it is technically a hypothetical combination of `EntityRef` and a `Query<&QueryState<D, F>>`  (a bit recursive there) but I didn't think that type existed so referenced the other one.

Mostly it was "if `Query` can share the entity how does that work" which I thought was what `EntityRef` was about "shared ownership of an `Entity`"